src/index.ts

```diff
@@ -348,12 +359,254 @@ async function certificateForImpl<
+ */
+export async function trustCertsOnRemote(
+  hostname: string,
+  port: number,
+  certPath: string,
+  renewalBufferInBusinessDays: number,
+  getRemoteCertsFunc = getRemoteCertificate
+): Promise<{ mustRenew: boolean }> {
+  if (!existsSync(certPath)) {
+    throw new Error(
+      `The certificate path ${certPath} to write the remote cert data on does not exist.`
+    );
+  }
+  // Get the remote certificate from the server
+  debug('getting cert from remote machine');
+  const certData = await getRemoteCertsFunc(hostname, port);
+  const mustRenew = shouldRenew(certData, renewalBufferInBusinessDays);
+  debug(`writing the certificate data onto local file path: ${certPath}`);
+
+  // Write the certificate data on this file.
+  writeFileSync(certPath, certData);
+
+  // Trust the remote cert on your local box
+  await currentPlatform.addToTrustStores(certPath);
+  debug('Certificate trusted successfully');
+  return { mustRenew };
+}
+/**
+ * Trust the remote hosts's certificate on local machine.
+ * This function would ssh into the remote host, get the certificate
+ * and trust the local machine from where this function is getting called from.
+ * @public
+ * @param hostname - hostname of the remote machine
+ * @param certPath - file path to store the cert
+ * @param param2 - TrustRemoteOptions options
+ */
+export async function trustRemoteMachine(
+  hostname: string,
+  certPath: string,
+  {
+    port = DEFAULT_REMOTE_PORT,
+    renewalBufferInBusinessDays = REMAINING_BUSINESS_DAYS_VALIDITY_BEFORE_RENEW,
+    logger
+  }: Partial<TrustRemoteOptions> = {}
+): Promise<{ mustRenew: boolean }> {
+  debug('fetching/generating domain cert data for connecting to remote');
+  const returnInfo = new Deferred<{ mustRenew: boolean }>();
+  const { cert, key } = await certificateFor(
+    'devcert-domain-cert',
+    [hostname],
+    {
+      skipHostsFile: true
+    }
+  );
+  const certData = cert.toString();
+  const keyData = key.toString();
+  _logOrDebug(logger, 'log', `Connecting to remote host ${hostname} via ssh`);
+  // Connect to remote box via ssh.
+  const child = execa.shell(
+    // @TODO Change this to npx
+    `ssh ${hostname} node devcert/bin/devcert.js remote --port=${port} --cert='${JSON.stringify(
+      certData
+    )}' --key='${JSON.stringify(keyData)}'`,
+    {
+      detached: false
+    }
+  );
+
+  // Error handling for missing handles on child process.
+  if (!child.stderr) {
+    throw new Error('Missing stderr on child process');
+  }
+  if (!child.stdout) {
+    throw new Error('Missing stdout on child process');
+  }
+
+  // Throw any error that might have occurred on the remote side.
+  child.stderr.on('data', (data: execa.StdIOOption) => {
+    if (data) {
+      const stdErrData = data.toString().trimRight();
+      debug(stdErrData);
+      if (stdErrData.toLowerCase().includes('error')) {
+        debug('Error thrown on the remote side. Closing Remote server');
+        closeRemoteServer(hostname, port);
```

Comment:
there is no real actionable item to do in case it fails here, may be we should throw something at the source itself? i.e. `closeRemoteServer`'s definition itself?